From ee580e013d08ad83456a3dc55700464cc98d04b6 Mon Sep 17 00:00:00 2001
From: Nico Weber <thakis@chromium.org>
Date: Tue, 18 May 2021 03:55:07 +0000
Subject: [PATCH] build: Convert third_party/node/node.gni to py3

Bug: 1205615
Change-Id: I9668aeb1edfbb67e0be6c0c2062f1c51b3f193f1
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/2896248
Auto-Submit: Nico Weber <thakis@chromium.org>
Commit-Queue: calamity <calamity@chromium.org>
Reviewed-by: dpapad <dpapad@chromium.org>
Reviewed-by: calamity <calamity@chromium.org>
Cr-Commit-Position: refs/heads/master@{#883788}
---
 .../browser/resources/tools/optimize_webui.py       | 13 +++++++------
 src/third_party/node/node.gni                       |  5 +----
 src/third_party/node/node.py                        |  3 ++-
 3 files changed, 10 insertions(+), 11 deletions(-)

diff --git a/src/chrome/browser/resources/tools/optimize_webui.py b/src/chrome/browser/resources/tools/optimize_webui.py
index f41a89bf81..32c4bba30e 100755
--- a/src/chrome/browser/resources/tools/optimize_webui.py
+++ b/src/chrome/browser/resources/tools/optimize_webui.py
@@ -258,12 +258,13 @@ def main(argv):
 
   # Output a manifest file that will be used to auto-generate a grd file later.
   if args.out_manifest:
-    manifest_data = {}
-    manifest_data['base_dir'] = '%s' % args.out_folder
-    manifest_data['files'] = manifest.keys()
-    manifest_file = open(
-        os.path.normpath(os.path.join(_CWD, args.out_manifest)), 'wb')
-    json.dump(manifest_data, manifest_file)
+    manifest_data = {
+      'base_dir': args.out_folder,
+      'files': list(manifest.keys()),
+    }
+    with open(os.path.normpath(os.path.join(_CWD, args.out_manifest)), 'w') \
+        as manifest_file:
+      json.dump(manifest_data, manifest_file)
 
   _update_dep_file(args.input, args, manifest)
 
diff --git a/src/third_party/node/node.gni b/src/third_party/node/node.gni
index fc2bc48797..81c4f2cb62 100644
--- a/src/third_party/node/node.gni
+++ b/src/third_party/node/node.gni
@@ -2,11 +2,8 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
-import("//build/config/python.gni")
-
 template("node") {
-  # TODO(crbug.com/1112471): Get this to run cleanly under Python 3.
-  python2_action(target_name) {
+  action(target_name) {
     forward_variables_from(invoker, "*", TESTONLY_AND_VISIBILITY)
     forward_variables_from(invoker, TESTONLY_AND_VISIBILITY)
 
diff --git a/src/third_party/node/node.py b/src/third_party/node/node.py
index 09662c4045..573987ba0c 100755
--- a/src/third_party/node/node.py
+++ b/src/third_party/node/node.py
@@ -27,7 +27,8 @@ def GetBinaryPath():
 def RunNode(cmd_parts, stdout=None):
   cmd = [GetBinaryPath()] + cmd_parts
   process = subprocess.Popen(
-      cmd, cwd=os.getcwd(), stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+      cmd, cwd=os.getcwd(), stdout=subprocess.PIPE, stderr=subprocess.PIPE,
+      universal_newlines=True)
   stdout, stderr = process.communicate()
 
   if process.returncode != 0:
